src/neural/loader.cc

```diff
@@ -100,6 +102,9 @@ FloatVectors LoadFloatsFromPbFile(const std::string& buffer) {
   FloatVectors vecs;
   net.ParseFromString(buffer);
 
+  if (net.magic() != kWeightMagic)
+    throw Exception("Invalid weight file");
```

Comment:
Maybe period at the end of a sentence? :)
Also maybe a bit more verbose: `Invalid weight file: bad header.`